web/styles/app_components.css

```diff
@@ -167,6 +167,15 @@ input::placeholder {
             background-color: hsl(0deg 0% 95%);
         }
 
+        &.add-subscription-button {
+            &:focus {
+                border: 1px solid;
+                border-color: var(
+                    --color-border-new-style-button-focus
+                ) !important;
```

Comment:
Is the `!important` necessary? It doesn't seem like it should be.

Also, why doesn't the native browser focus outline work for this button (or more generally, the focus outlines we have on other buttons in settings/modals)? I wonder if the `new-style` CSS class is a better solution than this custom code.

Reply:
Actually all buttons with class `new-style` do not have a outline on focus they just change background-color when focused (shown in code below). `!important` is necessary here to override this existing scenario in which the button only changes background color when in focus.
```css
    .new-style .button {
        &:hover,
        &:focus,
        &:active {
            background-color: hsl(0deg 0% 0% / 15%);
        }
    }
``` 
This can also be seen in the `unsubscribe` button and any other button with class new-style.
![Animation8](https://github.com/zulip/zulip/assets/91622060/26149d80-27fd-4e90-bbeb-c088266cac92)

Reply:
I think a better way to proceed here might be to zero out the focus styles, and instead use `:focus-visible`, which is the direction we should be working in on Zulip overall.

Based only on the evidence presented here, I don't understand why `border-color` would take `!important` if it's not set on `.new-style .button`.

Reply:
Okay I'll make that change.
Here in zulip.css
```css
.new-style {
    /* -- base button styling -- */
    .button {
        padding: 7px 14px;
        margin: 0;
        min-width: 130px;

        font-weight: 400;
        line-height: normal;
        text-align: center;

        background-color: hsl(0deg 0% 100%);
        color: inherit;
        outline: none;
        border: 1px solid hsl(0deg 0% 80%);
        border-radius: 2px;

        box-shadow: none;

        cursor: pointer;
        transition: all 0.2s ease;

        &:hover,
        &:focus {
            border-color: hsl(0deg 0% 60%);
        }
```
we have defined border-color when button is focused so as it override it I had to use `!important`.

Reply:
@karlstolley You mean like this?
![image](https://github.com/zulip/zulip/assets/91622060/7ba222e0-fc4a-4720-8d9c-21c61f08c378)
I am not sure what we're trying to achieve by doing this. I think I've got it wrong, can you please elaborate a bit more on this.

Reply:
Oh, I see. I was misunderstanding why `!important` was needed there, but having poked at this, it's the absolutely lamentable tangle of `:not()` selectors in `dark_theme.css`. We should probably put up an issue to fix that in a subsequent commit, with a note to remove this `!important` keyword, too.